grammar/rainerscript.c

```diff
@@ -2122,6 +2124,53 @@ doFuncCall(struct cnffunc *__restrict__ const func, struct svar *__restrict__ co
 		if(bMustFree) free(str);
 		varFreeMembers(&r[1]);
 		break;
+	case CNFFUNC_FORMAT_TIME: {
+		long long unixtime;
+		const int resMax = 64;
+		char   result[resMax];
+		char  *formatstr = NULL;
+
+		cnfexprEval(func->expr[0], &r[0], usrptr);
+		cnfexprEval(func->expr[1], &r[1], usrptr);
+
+		unixtime = var2Number(&r[0], &retval);
+
+		// Make sure that the timestamp we got can fit into
+		// time_t on older systems.
+		if (sizeof(time_t) == sizeof(int)) {
+			if (unixtime < INT_MIN || unixtime > INT_MAX) {
+				DBGPRINTF(
+					"Timestamp value %lld is out of range for this system (time_t is 32bits)!", unixtime
+				);
+				retval = 0;
+			}
+		}
+
+		// We want the string form too so we can return it as the
```

Comment:
I think we should (on the main issue) describe the semantics for the call. I don't think it makes sense to return the number if a date is expected, but that probably needs some discussion.

Reply:
Ok. Maybe I'm making assumptions about what the user would want to happen - rather than leaving it up to them to control the end result. I'm just loathe to discard the "real" value if something unexpected comes in.